internal/logger/logger_test.go

```diff
@@ -0,0 +1,300 @@
+	// Assert warning and error logs are logged when log level is WARNING.
+	var expected = []string{
+		"", "", "", textWarningString, textErrorString,
+	}
+	validateOutput(expected, output)
+}
+
+func (t *LoggerTest) TestTextFormatLogs_LogLevelINFO() {
+	// set log format to text
+	defaultLoggerFactory.format = "text"
+
+	output := fetchLogOutputForSpecifiedSeverityLevel(config.INFO, getTestLoggingFunctions())
+
+	// Assert info, warning & error logs are logged when log level is INFO.
+	var expected = []string{
+		"", "", textInfoString, textWarningString, textErrorString,
+	}
+	validateOutput(expected, output)
+}
+
+func (t *LoggerTest) TestTextFormatLogs_LogLevelDEBUG() {
+	// set log format to text
+	defaultLoggerFactory.format = "text"
+
+	output := fetchLogOutputForSpecifiedSeverityLevel(config.DEBUG, getTestLoggingFunctions())
+
+	// Assert debug, info, warning & error logs are logged when log level is DEBUG.
+	var expected = []string{
+		"", textDebugString, textInfoString, textWarningString, textErrorString,
+	}
+	validateOutput(expected, output)
+}
+
+func (t *LoggerTest) TestTextFormatLogs_LogLevelTRACE() {
+	// set log format to text
+	defaultLoggerFactory.format = "text"
+
+	output := fetchLogOutputForSpecifiedSeverityLevel(config.TRACE, getTestLoggingFunctions())
+
+	// Assert all logs are logged when log level is TRACE.
+	var expected = []string{
+		textTraceString, textDebugString, textInfoString, textWarningString, textErrorString,
+	}
+	validateOutput(expected, output)
+}
+
+func (t *LoggerTest) TestJSONFormatLogs_LogLevelOFF() {
+	// set log format to json
+	defaultLoggerFactory.format = "json"
+
+	outputs := fetchLogOutputForSpecifiedSeverityLevel(config.OFF, getTestLoggingFunctions())
+
+	// Assert that nothing is logged when log level is OFF.
+	for _, output := range outputs {
+		AssertEq("", output)
+	}
+}
+
+func (t *LoggerTest) TestJSONFormatLogs_LogLevelERROR() {
+	// set log format to json
+	defaultLoggerFactory.format = "json"
+
+	output := fetchLogOutputForSpecifiedSeverityLevel(config.ERROR, getTestLoggingFunctions())
+
+	// Assert only error logs are logged when log level is ERROR.
+	var expected = []string{
+		"", "", "", "", jsonErrorString,
+	}
+	validateOutput(expected, output)
+}
+
+func (t *LoggerTest) TestJSONFormatLogs_LogLevelWARNING() {
+	// set log format to json
+	defaultLoggerFactory.format = "json"
+
+	output := fetchLogOutputForSpecifiedSeverityLevel(config.WARNING, getTestLoggingFunctions())
+
+	// Assert warning and error logs are logged when log level is WARNING.
+	var expected = []string{
+		"", "", "", jsonWarningString, jsonErrorString,
+	}
+	validateOutput(expected, output)
+}
+
+func (t *LoggerTest) TestJSONFormatLogs_LogLevelINFO() {
+	// set log format to json
+	defaultLoggerFactory.format = "json"
+
+	output := fetchLogOutputForSpecifiedSeverityLevel(config.INFO, getTestLoggingFunctions())
+
+	// Assert info, warning & error logs are logged when log level is INFO.
+	var expected = []string{
```

Comment:
can you extract all this common logic into seperate method and pass 3 methods -> format, config level, expected array

Reply:
done